airflow/models/pool.py

```diff
@@ -161,12 +161,13 @@ def occupied_slots(self, session: Session):
         """
         from airflow.models.taskinstance import TaskInstance  # Avoid circular import
 
-        return (
+        return int(
             session.query(func.sum(TaskInstance.pool_slots))
             .filter(TaskInstance.pool == self.pool)
             .filter(TaskInstance.state.in_(list(EXECUTION_STATES)))
             .scalar()
-        ) or 0
+            or 0
```

Comment:
Interesting I thought `.scalar()` should take care of it based on https://stackoverflow.com/a/53431701/5691525 too

Reply:
This is the implementation, there's no magic at all. I guess the SO answer is inaccurate.

https://github.com/sqlalchemy/sqlalchemy/blob/b703a9c1be728077e4709c377b0b29a94c052a0b/lib/sqlalchemy/orm/query.py#L2818-L2824